### PR TITLE
Nasfpn

### PR DESCRIPTION
**Pull request by huangzehao**
- New Retinanet setting follow https://github.com/tensorflow/tpu/tree/master/models/official/retinanet
- NASFPN

### CODE REVIEW

**Comment by RogerChern** [comment]
Update model_zoo and readme -> recent updates accordingly.

File: config/NASFPN/retina_r50v1b_fpn_640640_25epoch.py [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+            ratio = (0.5, 1.0, 2.0)
+            stride = (8, 16, 32, 64, 128)
+            image_anchor = None
+
+        class head:
+            conv_channel = 256
+            mean = None
+            std = None
+
+        class proposal:
```
Comment (RogerChern) [inline config/NASFPN/retina_r50v1b_fpn_640640_25epoch.py]: If a whole sub config is not used, just remove it.

File: models/NASFPN/builder.py [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+def reluconvbn(data, num_filter, init, norm, name, prefix):
+    """
+    :param data: data
+    :param num_filter: number of convolution filter
+    :param init: init method of conv weight
+    :param norm: normalizer
+    :param name: name
+    :return: relu-3x3conv-bn
+    """
+    data = mx.sym.Activation(data, name=name+'_relu', act_type='relu')
```
Comment (RogerChern) [inline models/NASFPN/builder.py]: name=name + '_relu'

File: models/NASFPN/builder.py [hunk 3]
```diff
@@ -0,0 +1,522 @@
...
+    bias = mx.sym.var(name=prefix + name + "_bias")
+    data = mx.sym.Convolution(data, name=prefix + name, weight=weight, bias=bias, num_filter=num_filter, kernel=(3, 3), pad=(1, 1), stride=(1, 1))
+    data = norm(data, name=name+'_bn')
+    return data
+
+
+class NASFPNNeck(Neck):
+    def __init__(self, pNeck):
+        super(NASFPNNeck, self).__init__(pNeck)
+        self.p = pNeck
```
Comment (RogerChern) [inline models/NASFPN/builder.py]: DO NOT do this. This will void the patch_config. Just let the base class handle the `self.p` assginment.

**Comment by huangzehao** [comment]
@RogerChern 
Ready to merge.

**Pull request merged**